### ISSUE TITLE
Minimal tensorflow version

### ISSUE THREAD

**Issue by nathanin** [xu-ji/IIC#2]
Hi, over the past few days I put together a minimal tensorflow version to study your information based clustering algorithm.

https://github.com/nathanin/IIC/tree/master/tensorflow 

The MNIST example can be run in the `tf_cluster.py` script. It wants a directory called `pointcloud` to draw the outputs to. 

The loss functions are tested to give nearly identical results in tensorflow and pytorch (see the readme there). Still, I'm having trouble training very robust clustering as shown in the MNIST example. The differences would be in the conv net architecture, and the sub-head & auxiliary overclustering heads, unfortunately I'm not fluent in pytorch and it's hard for me to parse the reference package. Currently I only have the MNIST example in place, with regular python iteration to feed the data, instead of a tf.data.dataset. This results in some slow training when image augmentations like flips and rotates are turned on. 

I'll keep working on this and make updates to my fork linked above.

Best,

**Comment by xu-ji** [comment]
Great. Yes, the architecture, auxiliary overclustering, and transforms used all make a difference. Multiple sub-heads does help but is secondary to those. 

Auxiliary overclustering is easy to implement, it's literally a second output layer with the same format and loss as the main one (but bigger number of channels). 

I give the command used to train my MNIST model [here](https://github.com/xu-ji/IIC/blob/master/examples/commands.txt#L30). Tracing this in the code will show you the exact architecture, transforms etc used.

**Comment by xu-ji** [comment]
I've added a link to your repo in the main readme.
